tools/garaga_rs/src/extf_mul.rs

```diff
@@ -0,0 +1,155 @@
+use crate::ecip::polynomial::{pad_with_zero_coefficients_to_length, Polynomial};
+use lambdaworks_math::{
+    elliptic_curve::short_weierstrass::curves::{
+        bls12_381::field_extension::{BLS12381PrimeField, BLS12381_PRIME_FIELD_ORDER},
+        bn_254::field_extension::{BN254PrimeField, BN254_PRIME_FIELD_ORDER},
+    },
+    field::element::FieldElement,
+    traits::ByteConversion,
+    unsigned_integer::element::{U256, U384},
+};
+use num_bigint::BigUint;
+
+// irreducible polynomial coefficients
+const IPC: [[&[i8]; 2]; 2] = [
+    // curve 0: BN254
+    [
+        &[82, 0, 0, -18, 0, 0, 1],                   // 6
+        &[82, 0, 0, 0, 0, 0, -18, 0, 0, 0, 0, 0, 1], // 12
+    ],
+    // curve 1: BLS12-381
+    [
+        &[2, 0, 0, -2, 0, 0, 1],                   // 6
+        &[2, 0, 0, 0, 0, 0, -2, 0, 0, 0, 0, 0, 1], // 12
+    ],
+];
+
+// Returns (Q(X), R(X)) such that Π(Pi)(X) = Q(X) * P_irr(X) + R(X), for a given curve and extension degree.
+// R(X) is the result of the multiplication in the extension field.
+// Q(X) is used for verification.
+pub fn nondeterministic_extension_field_mul_divmod(
+    curve_id: usize,
+    ext_degree: usize,
+    list_coeffs: Vec<Vec<Vec<u8>>>,
+) -> Result<(Vec<BigUint>, Vec<BigUint>), String> {
```

Comment:
Can you make this inner function take a list of `polynomial<F>` as input, and implement some trait (or something else) that binds 6 or 12 with a constant irreducible polynomial depending on `<F>` ? 

This will be easier because we're going to re-use this function many times in the multi miller loop algorithms and we want to keep polynomials as input/output.  

The mapping between a curve id and <F> should be done in the python binding only.